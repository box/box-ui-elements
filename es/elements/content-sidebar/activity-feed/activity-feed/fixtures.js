export const replies = [{
  created_at: '2020-09-20T00:00:00Z',
  created_by: {
    id: '1',
    login: 'johndoe',
    name: 'John Doe',
    type: 'user'
  },
  id: '2485934',
  message: 'First reply',
  tagged_message: 'First reply',
  modified_at: '2020-09-20T00:00:00Z',
  modified_by: {
    id: '1',
    login: 'johndoe',
    name: 'John Doe',
    type: 'user'
  },
  permissions: {
    can_delete: true,
    can_edit: true,
    can_reply: false,
    can_resolve: true
  },
  type: 'comment'
}, {
  created_at: '2020-09-20T00:00:00Z',
  created_by: {
    id: '1',
    login: 'johndoe',
    name: 'John Doe',
    type: 'user'
  },
  id: '24835935',
  message: 'Last reply',
  tagged_message: 'Last reply',
  modified_at: '2020-09-20T00:00:00Z',
  modified_by: {
    id: '1',
    login: 'johndoe',
    name: 'John Doe',
    type: 'user'
  },
  permissions: {
    can_delete: true,
    can_edit: true,
    can_reply: false,
    can_resolve: true
  },
  type: 'comment'
}];
export default {
  replies
};
//# sourceMappingURL=fixtures.js.map