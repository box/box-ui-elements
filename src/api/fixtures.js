// @flow
import { TASK_NEW_NOT_STARTED } from '../constants';

export const task = {
    created_by: {
        type: 'task_collaborator',
        target: { name: 'Jay-Z', id: '100' },
        id: '000',
        role: 'CREATOR',
        status: TASK_NEW_NOT_STARTED,
    },
    created_at: '2019-01-01',
    due_at: '2019-02-02',
    id: '0',
    name: 'task message',
    type: 'task',
    assigned_to: {
        entries: [
            {
                id: '1',
                target: { name: 'Beyonce', id: '2', avatar_url: '', type: 'user' },
                status: TASK_NEW_NOT_STARTED,
                permissions: {
                    can_delete: false,
                    can_update: false,
                },
                role: 'ASSIGNEE',
                type: 'task_collaborator',
            },
        ],
        limit: 10,
        next_marker: null,
    },
    permissions: {
        can_update: false,
        can_delete: false,
        can_create_task_collaborator: false,
        can_create_task_link: false,
    },
    task_links: {
        entries: [
            {
                id: '03',
                type: 'task_link',
                target: {
                    type: 'file',
                    id: '4',
                },
                permissions: {
                    can_delete: false,
                    can_update: false,
                },
            },
        ],
        limit: 1,
        next_marker: null,
    },
    status: TASK_NEW_NOT_STARTED,
};

export const threadedComments = [
    {
        created_at: '1970-01-01T00:00:00.001Z',
        created_by: {
            id: '10',
            type: 'user',
            name: 'u1_name',
            login: 'u1@box.com',
        },
        id: '20',
        item: {
            type: 'file',
            id: 'f1',
        },
        message: '@[111:Aaron Levie] these tigers are cool!',
        modified_at: '1970-01-01T00:00:00.001Z',
        type: 'comment',
        parent: null,
        permissions: {
            can_delete: false,
            can_edit: false,
            can_reply: true,
            can_resolve: true,
        },
        replies: [
            {
                created_at: '1970-01-01T00:00:00.002Z',
                created_by: {
                    id: '11',
                    type: 'user',
                    name: 'u2_name',
                    login: 'u2@box.com',
                },
                id: '21',
                item: {
                    type: 'file',
                    id: 'f1',
                },
                message: '@[u1:Mateusz Mamczarz] Yes, they really are!',
                modified_at: '1970-01-01T00:00:00.002Z',
                type: 'comment',
                parent: {
                    id: 'c1',
                    type: 'comment',
                },
                permissions: {
                    can_delete: false,
                    can_edit: false,
                    can_reply: false,
                    can_resolve: true,
                },
                replies: [],
                total_reply_count: 0,
                status: 'open',
            },
        ],
        total_reply_count: 1,
        status: 'open',
    },
    {
        created_at: '1970-01-01T00:00:00.003Z',
        created_by: {
            id: '12',
            type: 'user',
            name: 'u3_name',
            login: 'u3@box.com',
        },
        id: '21',
        item: {
            type: 'file',
            id: 'f1',
        },
        message: 'Test message',
        modified_at: '1970-01-01T00:00:00.003Z',
        type: 'comment',
        parent: null,
        permissions: {
            can_delete: false,
            can_edit: false,
            can_reply: true,
            can_resolve: true,
        },
        replies: [],
        total_reply_count: 0,
        status: 'open',
    },
];

export default {
    task,
    threadedComments,
};
