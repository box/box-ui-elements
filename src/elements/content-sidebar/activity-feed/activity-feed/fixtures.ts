export const replies = [
    {
        created_at: '2020-09-20T00:00:00Z',
        created_by: {
            id: '1',
            login: 'johndoe',
            name: 'John Doe',
            type: 'user' as const,
        },
        id: '2485934',
        message: 'First reply',
        tagged_message: 'First reply',
        modified_at: '2020-09-20T00:00:00Z',
        modified_by: {
            id: '1',
            login: 'johndoe',
            name: 'John Doe',
            type: 'user' as const,
        },
        permissions: {
            can_delete: true,
            can_edit: true,
            can_reply: false,
            can_resolve: true,
        },
        type: 'comment' as const,
    },
    {
        created_at: '2020-09-20T00:00:00Z',
        created_by: {
            id: '1',
            login: 'johndoe',
            name: 'John Doe',
            type: 'user' as const,
        },
        id: '24835935',
        message: 'Last reply',
        tagged_message: 'Last reply',
        modified_at: '2020-09-20T00:00:00Z',
        modified_by: {
            id: '1',
            login: 'johndoe',
            name: 'John Doe',
            type: 'user' as const,
        },
        permissions: {
            can_delete: true,
            can_edit: true,
            can_reply: false,
            can_resolve: true,
        },
        type: 'comment' as const,
    },
];

export default { replies };
