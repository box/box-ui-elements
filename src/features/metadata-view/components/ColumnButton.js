// @flow
import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import classNames from 'classnames';
import uniqueId from 'lodash/uniqueId';

import { Flyout, Overlay } from 'components/flyout';
import Button from 'components/button/Button';
import Checkbox from 'components/checkbox/Checkbox';
import DraggableList from 'components/draggable-list';
import PortaledDraggableListItem from 'components/draggable-list/PortaledDraggableListItem';
import MenuToggle from 'components/dropdown-menu/MenuToggle';
import PrimaryButton from 'components/primary-button/PrimaryButton';
import reorder from 'components/draggable-list/draggable-list-utils/reorder';
import IconMetadataColumns from '../../../icons/metadata-view/IconMetadataColumns';

import type { Template } from '../../metadata-instance-editor/flowTypes';
import type { ColumnType } from '../flowTypes';

import messages from '../messages';

type State = {
    isColumnMenuOpen: boolean,
    listId: string,
    unsavedVisibleColumns: Array<ColumnType>,
};

type Props = {
    onColumnChange?: Function,
    template?: Template,
    visibleColumns?: Array<ColumnType>,
};

class ColumnButton extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);
        this.state = {
            isColumnMenuOpen: false,
            listId: uniqueId(),
            unsavedVisibleColumns: props.visibleColumns ? props.visibleColumns.slice(0) : [],
        };
    }

    onDragEnd = (sourceIndex: number, destinationIndex: number) => {
        const { unsavedVisibleColumns } = this.state;
        const columns = reorder(unsavedVisibleColumns, sourceIndex, destinationIndex);
        this.setState({
            unsavedVisibleColumns: columns.slice(0),
        });
    };

    onClose = () => {
        this.setState({
            isColumnMenuOpen: false,
        });
    };

    onOpen = () => {
        const { visibleColumns } = this.props;
        this.setState({
            isColumnMenuOpen: true,
            unsavedVisibleColumns: visibleColumns,
        });
    };

    toggleColumnButton = () => {
        this.setState({ isColumnMenuOpen: !this.state.isColumnMenuOpen });
    };

    updateUnsavedVisibleColumns = (column: ColumnType) => {
        const { unsavedVisibleColumns } = this.state;

        const unsavedVisibleColumnsCopy = unsavedVisibleColumns.slice(0);

        const newColumn = { ...column, isChecked: !column.isChecked };

        const foundIndex = unsavedVisibleColumnsCopy.findIndex(originalColumn => originalColumn.id === column.id);

        unsavedVisibleColumnsCopy[foundIndex] = newColumn;

        this.setState({
            unsavedVisibleColumns: unsavedVisibleColumnsCopy,
        });
    };

    applyFilters = () => {
        const { onColumnChange } = this.props;
        const { unsavedVisibleColumns } = this.state;
        if (onColumnChange) {
            onColumnChange(unsavedVisibleColumns);
        }
    };

    getNumberOfHiddenColumns = () => {
        const { visibleColumns } = this.props;

        return visibleColumns
            ? visibleColumns.reduce((total, column) => {
                  if (!column.isChecked) {
                      return total + 1;
                  }
                  return total;
              }, 0)
            : 0;
    };

    render() {
        const { template } = this.props;
        const { listId, unsavedVisibleColumns } = this.state;

        const numberOfHiddenColumns = this.getNumberOfHiddenColumns();

        const buttonClasses = classNames('query-bar-button', numberOfHiddenColumns !== 0 ? 'is-active' : '');

        return (
            <Flyout
                className="query-bar-column-dropdown-flyout"
                closeOnClick
                closeOnClickOutside
                onClose={this.onClose}
                onOpen={this.onOpen}
                position="bottom-right"
            >
                <Button
                    className={buttonClasses}
                    isDisabled={template === undefined}
                    onClick={this.toggleColumnButton}
                    type="button"
                >
                    <MenuToggle>
                        <IconMetadataColumns />
                        <span className="button-label">
                            {numberOfHiddenColumns === 0 ? (
                                <FormattedMessage {...messages.columnsButtonText} />
                            ) : (
                                <FormattedMessage
                                    {...messages.columnsHiddenButtonText}
                                    values={{
                                        number: numberOfHiddenColumns,
                                    }}
                                />
                            )}
                        </span>
                    </MenuToggle>
                </Button>

                <Overlay>
                    <div className="column-button-dropdown">
                        <div className="column-button-dropdown-header">
                            <DraggableList
                                className="draggable-list-example"
                                listId={listId}
                                onDragEnd={this.onDragEnd}
                            >
                                {unsavedVisibleColumns.map((item, index) => {
                                    // We skip the first index as it's always the name column, and that can't be modified
                                    if (index !== 0) {
                                        return (
                                            <PortaledDraggableListItem
                                                id={item.id}
                                                index={index}
                                                isDraggableViaHandle
                                                key={index}
                                            >
                                                <Checkbox
                                                    isChecked={item.isChecked}
                                                    label={item.label}
                                                    name={item.label}
                                                    onChange={() => this.updateUnsavedVisibleColumns(item)}
                                                />
                                            </PortaledDraggableListItem>
                                        );
                                    }
                                    return null;
                                })}
                            </DraggableList>
                        </div>
                        <div className="column-button-dropdown-footer">
                            <PrimaryButton type="button" onClick={this.applyFilters}>
                                <FormattedMessage {...messages.applyFiltersButtonText} />
                            </PrimaryButton>
                        </div>
                    </div>
                </Overlay>
            </Flyout>
        );
    }
}

export default ColumnButton;
