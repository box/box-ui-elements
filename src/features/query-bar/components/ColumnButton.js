// @flow
import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import classNames from 'classnames';

import { Flyout, Overlay } from '../../../components/flyout';
import Button from '../../../components/button/Button';
import MenuToggle from '../../../components/dropdown-menu/MenuToggle';
import IconMetadataColumns from '../../../icons/metadata-view/IconMetadataColumns';
import ColumnButtonOverlay from './ColumnButtonOverlay';

import type { ColumnType } from '../flowTypes';
import type { MetadataTemplate } from '../../../common/types/metadata';

import messages from '../messages';

type State = {
    isColumnMenuOpen: boolean,
};

type Props = {
    columns?: Array<ColumnType>,
    onColumnChange?: (columnTypes: Array<ColumnType>) => void,
    template?: MetadataTemplate,
};

class ColumnButton extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);
        this.state = {
            isColumnMenuOpen: false,
        };
    }

    onClose = () => {
        this.setState({
            isColumnMenuOpen: false,
        });
    };

    onOpen = () => {
        this.setState({
            isColumnMenuOpen: true,
        });
    };

    toggleColumnButton = () => {
        this.setState({ isColumnMenuOpen: !this.state.isColumnMenuOpen });
    };

    getNumberOfHiddenColumns = () => {
        const { columns } = this.props;

        return columns
            ? columns.reduce((total, column) => {
                  if (!column.isShown) {
                      return total + 1;
                  }
                  return total;
              }, 0)
            : 0;
    };

    render() {
        const { template, columns, onColumnChange } = this.props;
        const { isColumnMenuOpen } = this.state;
        const numberOfHiddenColumns = this.getNumberOfHiddenColumns();

        const buttonClasses = classNames('query-bar-button', numberOfHiddenColumns !== 0 ? 'is-active' : '');

        let columnsButtonText;
        if (numberOfHiddenColumns === 0) {
            columnsButtonText = <FormattedMessage {...messages.columnsButtonText} />;
        } else {
            columnsButtonText = (
                <FormattedMessage
                    values={{
                        count: numberOfHiddenColumns,
                    }}
                    {...messages.columnsHiddenButtonText}
                />
            );
        }

        return (
            <Flyout
                className="query-bar-column-dropdown-flyout"
                closeOnClick
                closeOnClickOutside
                onClose={this.onClose}
                onOpen={this.onOpen}
                position="bottom-right"
            >
                <Button
                    className={buttonClasses}
                    isDisabled={template === undefined}
                    onClick={this.toggleColumnButton}
                    type="button"
                >
                    <MenuToggle>
                        <IconMetadataColumns />
                        <span className="button-label">{columnsButtonText}</span>
                    </MenuToggle>
                </Button>

                <Overlay>
                    {isColumnMenuOpen ? (
                        <ColumnButtonOverlay columns={columns} onColumnChange={onColumnChange} />
                    ) : (
                        <div />
                    )}
                </Overlay>
            </Flyout>
        );
    }
}

export default ColumnButton;
