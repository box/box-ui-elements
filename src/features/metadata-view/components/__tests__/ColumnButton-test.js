import * as React from 'react';

import { visibleColumns } from '../fixtures';
import ColumnButton from '../ColumnButton';

describe('features/metadata-view/components/ColumnButton', () => {
    const getWrapper = (props = {}) => {
        return shallow(<ColumnButton onColumnChange={jest.fn()} visibleColumns={visibleColumns} {...props} />);
    };

    describe('render', () => {
        test('should render ColumnButton default state', () => {
            const wrapper = getWrapper();
            expect(wrapper).toMatchSnapshot();
        });

        test('should render ColumnButton with template passed in', () => {
            const template = {
                'Vendor Name': {
                    operators: ['is', 'is greater than', 'is less than', 'is not', 'is blank', 'matches any'],
                    values: ['Google', 'Apple', 'Facebook'],
                },
                'Expiration Month': {
                    operators: ['is', 'is greater than', 'is less than', 'is not'],
                    values: ['August 2018', 'September 2018', 'October 2018'],
                },
                'File Type': {
                    operators: ['is', 'is not'],
                    values: ['.docx', '.mp3', 'mp4'],
                },
            };

            const wrapper = getWrapper({ template });
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('onDragEnd()', () => {
        [
            {
                description: 'Should update state with new ordering',
                sourceIndex: 0,
                destinationIndex: 2,
                updatedColumns: [
                    {
                        displayName: 'Vendor Name',
                        id: 'item_28',
                        isChecked: true,
                        key: 'vendor name',
                        label: 'Vendor Name',
                    },
                    {
                        displayName: 'Updated',
                        id: 'item_26',
                        isChecked: true,
                        key: 'updated',
                        label: 'Updated',
                    },
                    {
                        displayName: 'Size',
                        id: 'item_27',
                        isChecked: true,
                        key: 'size',
                        label: 'Size',
                    },
                    {
                        displayName: 'Contract Value',
                        id: 'item_29',
                        isChecked: true,
                        key: 'contract value',
                        label: 'Contract Value',
                    },
                    {
                        displayName: 'Expiration Month',
                        id: 'item_30',
                        isChecked: true,
                        key: 'expiration month',
                        label: 'Expiration Month',
                    },
                    {
                        displayName: 'Country',
                        id: 'item_31',
                        isChecked: true,
                        key: 'country',
                        label: 'Country',
                    },
                    {
                        displayName: 'State',
                        id: 'item_32',
                        isChecked: true,
                        key: 'state',
                        label: 'State',
                    },
                    {
                        displayName: 'Function',
                        id: 'item_33',
                        isChecked: true,
                        key: 'function',
                        label: 'Function',
                    },
                ],
            },
            {
                description: 'Should not update state due to no destinationIndex',
                sourceIndex: 0,
                updatedColumns: [
                    {
                        displayName: 'Size',
                        id: 'item_27',
                        isChecked: true,
                        key: 'size',
                        label: 'Size',
                    },
                    {
                        displayName: 'Vendor Name',
                        id: 'item_28',
                        isChecked: true,
                        key: 'vendor name',
                        label: 'Vendor Name',
                    },
                    {
                        displayName: 'Updated',
                        id: 'item_26',
                        isChecked: true,
                        key: 'updated',
                        label: 'Updated',
                    },
                    {
                        displayName: 'Contract Value',
                        id: 'item_29',
                        isChecked: true,
                        key: 'contract value',
                        label: 'Contract Value',
                    },
                    {
                        displayName: 'Expiration Month',
                        id: 'item_30',
                        isChecked: true,
                        key: 'expiration month',
                        label: 'Expiration Month',
                    },
                    {
                        displayName: 'Country',
                        id: 'item_31',
                        isChecked: true,
                        key: 'country',
                        label: 'Country',
                    },
                    {
                        displayName: 'State',
                        id: 'item_32',
                        isChecked: true,
                        key: 'state',
                        label: 'State',
                    },
                    {
                        displayName: 'Function',
                        id: 'item_33',
                        isChecked: true,
                        key: 'function',
                        label: 'Function',
                    },
                ],
            },
        ].forEach(({ description, sourceIndex, destinationIndex, updatedColumns }) => {
            test(`${description}`, () => {
                const wrapper = getWrapper();
                wrapper.instance().onDragEnd(sourceIndex, destinationIndex);

                expect(wrapper.state('unsavedVisibleColumns')).toEqual(updatedColumns);
            });
        });
    });

    describe('onClose()', () => {
        [
            {
                description: 'Should update state with new ordering',
                updatedState: {
                    isColumnMenuOpen: false,
                },
            },
        ].forEach(({ description, updatedState }) => {
            test(`${description}`, () => {
                const wrapper = getWrapper();
                wrapper.instance().onClose();

                expect(wrapper.state('isColumnMenuOpen')).toEqual(updatedState.isColumnMenuOpen);
            });
        });
    });

    describe('onOpen()', () => {
        [
            {
                description: 'Should update state with new ordering',
                updatedState: {
                    isColumnMenuOpen: true,
                },
            },
        ].forEach(({ description, updatedState }) => {
            test(`${description}`, () => {
                const wrapper = getWrapper();
                wrapper.instance().onOpen();

                expect(wrapper.state('isColumnMenuOpen')).toEqual(updatedState.isColumnMenuOpen);
            });
        });
    });

    describe('toggleColumnButton()', () => {
        [
            {
                description: 'Should update state with new ordering',
                updatedState: {
                    isColumnMenuOpen: true,
                },
            },
        ].forEach(({ description, updatedState }) => {
            test(`${description}`, () => {
                const wrapper = getWrapper();
                wrapper.instance().toggleColumnButton();

                expect(wrapper.state('isColumnMenuOpen')).toEqual(updatedState.isColumnMenuOpen);
            });
        });
    });

    describe('applyFilters()', () => {
        [
            {
                description: 'Should apply filters to parent state',
                unsavedVisibleColumns: [
                    {
                        id: 'item_27',
                        isChecked: true,
                        label: 'Size',
                        key: 'size',
                        displayName: 'Size',
                    },
                    {
                        id: 'item_28',
                        isChecked: true,
                        label: 'Vendor Name',
                        key: 'vendor name',
                        displayName: 'Vendor Name',
                    },
                    {
                        id: 'item_26',
                        isChecked: true,
                        label: 'Updated',
                        key: 'updated',
                        displayName: 'Updated',
                    },
                    {
                        id: 'item_29',
                        isChecked: true,
                        label: 'Contract Value',
                        key: 'contract value',
                        displayName: 'Contract Value',
                    },
                    {
                        id: 'item_30',
                        isChecked: true,
                        label: 'Expiration Month',
                        key: 'expiration month',
                        displayName: 'Expiration Month',
                    },
                    {
                        id: 'item_31',
                        isChecked: true,
                        label: 'Country',
                        key: 'country',
                        displayName: 'Country',
                    },
                    {
                        id: 'item_32',
                        isChecked: true,
                        label: 'State',
                        key: 'state',
                        displayName: 'State',
                    },
                    {
                        id: 'item_33',
                        isChecked: true,
                        label: 'Function',
                        key: 'function',
                        displayName: 'Function',
                    },
                ],
            },
        ].forEach(({ description, unsavedVisibleColumns }) => {
            test(`${description}`, () => {
                const wrapper = getWrapper();

                wrapper.setState({
                    unsavedVisibleColumns,
                });
                wrapper.instance().applyFilters();
            });
        });
    });

    describe('getNumberOfHiddenColumns()', () => {
        [
            {
                description: 'Should return the number of hidden columns',
                numberOfHiddenColumns: 0,
            },
        ].forEach(({ description, numberOfHiddenColumns }) => {
            test(`${description}`, () => {
                const wrapper = getWrapper();
                const result = wrapper.instance().getNumberOfHiddenColumns();

                expect(result).toEqual(numberOfHiddenColumns);
            });
        });
    });

    describe('updateUnsavedVisibleColumns()', () => {
        [
            {
                description: 'Should update state with unsavedVisibleColumns',
                numberOfHiddenColumns: 0,
                column: {
                    id: 'item_27',
                    isChecked: true,
                    label: 'Size',
                    key: 'size',
                    displayName: 'Size',
                },
                unsavedVisibleColumns: [
                    {
                        id: 'item_27',
                        isChecked: true,
                        label: 'Size',
                        key: 'size',
                        displayName: 'Size',
                    },
                    {
                        id: 'item_28',
                        isChecked: true,
                        label: 'Vendor Name',
                        key: 'vendor name',
                        displayName: 'Vendor Name',
                    },
                    {
                        id: 'item_26',
                        isChecked: true,
                        label: 'Updated',
                        key: 'updated',
                        displayName: 'Updated',
                    },
                    {
                        id: 'item_29',
                        isChecked: true,
                        label: 'Contract Value',
                        key: 'contract value',
                        displayName: 'Contract Value',
                    },
                    {
                        id: 'item_30',
                        isChecked: true,
                        label: 'Expiration Month',
                        key: 'expiration month',
                        displayName: 'Expiration Month',
                    },
                    {
                        id: 'item_31',
                        isChecked: true,
                        label: 'Country',
                        key: 'country',
                        displayName: 'Country',
                    },
                    {
                        id: 'item_32',
                        isChecked: true,
                        label: 'State',
                        key: 'state',
                        displayName: 'State',
                    },
                    {
                        id: 'item_33',
                        isChecked: true,
                        label: 'Function',
                        key: 'function',
                        displayName: 'Function',
                    },
                ],
                updatedUnsavedVisibleColumns: [
                    {
                        id: 'item_27',
                        isChecked: false,
                        label: 'Size',
                        key: 'size',
                        displayName: 'Size',
                    },
                    {
                        id: 'item_28',
                        isChecked: true,
                        label: 'Vendor Name',
                        key: 'vendor name',
                        displayName: 'Vendor Name',
                    },
                    {
                        id: 'item_26',
                        isChecked: true,
                        label: 'Updated',
                        key: 'updated',
                        displayName: 'Updated',
                    },
                    {
                        id: 'item_29',
                        isChecked: true,
                        label: 'Contract Value',
                        key: 'contract value',
                        displayName: 'Contract Value',
                    },
                    {
                        id: 'item_30',
                        isChecked: true,
                        label: 'Expiration Month',
                        key: 'expiration month',
                        displayName: 'Expiration Month',
                    },
                    {
                        id: 'item_31',
                        isChecked: true,
                        label: 'Country',
                        key: 'country',
                        displayName: 'Country',
                    },
                    {
                        id: 'item_32',
                        isChecked: true,
                        label: 'State',
                        key: 'state',
                        displayName: 'State',
                    },
                    {
                        id: 'item_33',
                        isChecked: true,
                        label: 'Function',
                        key: 'function',
                        displayName: 'Function',
                    },
                ],
            },
        ].forEach(({ description, column, unsavedVisibleColumns, updatedUnsavedVisibleColumns }) => {
            test(`${description}`, () => {
                const wrapper = getWrapper();

                wrapper.setState({
                    unsavedVisibleColumns,
                });

                wrapper.instance().updateUnsavedVisibleColumns(column);

                expect(wrapper.state('unsavedVisibleColumns')).toEqual(updatedUnsavedVisibleColumns);
            });
        });
    });
});
