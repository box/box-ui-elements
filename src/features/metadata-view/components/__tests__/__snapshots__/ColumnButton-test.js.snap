// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features/metadata-view/components/ColumnButton render should render ColumnButton default state 1`] = `
<Flyout
  className="query-bar-column-dropdown-flyout"
  closeOnClick={true}
  closeOnClickOutside={true}
  closeOnWindowBlur={false}
  constrainToScrollParent={true}
  constrainToWindow={false}
  isVisibleByDefault={false}
  onClose={[Function]}
  onOpen={[Function]}
  openOnHover={false}
  openOnHoverDelayTimeout={300}
  portaledClasses={Array []}
  position="bottom-right"
>
  <Button
    className="query-bar-button"
    isDisabled={true}
    onClick={[Function]}
    type="button"
  >
    <MenuToggle>
      <IconMetadataColumns />
      <span
        className="button-label"
      >
        <FormattedMessage
          defaultMessage="Columns"
          id="boxui.metadataView.columnsButtonText"
        />
      </span>
    </MenuToggle>
  </Button>
  <Overlay>
    <div
      className="column-button-dropdown"
    >
      <div
        className="column-button-dropdown-header"
      >
        <DraggableList
          className="draggable-list-example"
          listId="1"
          onDragEnd={[Function]}
        >
          <PortaledDraggableListItem
            id="item_28"
            index={1}
            isDraggableViaHandle={true}
            key="1"
          >
            <Checkbox
              isChecked={true}
              label="Vendor Name"
              name="Vendor Name"
              onChange={[Function]}
            />
          </PortaledDraggableListItem>
          <PortaledDraggableListItem
            id="item_26"
            index={2}
            isDraggableViaHandle={true}
            key="2"
          >
            <Checkbox
              isChecked={true}
              label="Updated"
              name="Updated"
              onChange={[Function]}
            />
          </PortaledDraggableListItem>
          <PortaledDraggableListItem
            id="item_29"
            index={3}
            isDraggableViaHandle={true}
            key="3"
          >
            <Checkbox
              isChecked={true}
              label="Contract Value"
              name="Contract Value"
              onChange={[Function]}
            />
          </PortaledDraggableListItem>
          <PortaledDraggableListItem
            id="item_30"
            index={4}
            isDraggableViaHandle={true}
            key="4"
          >
            <Checkbox
              isChecked={true}
              label="Expiration Month"
              name="Expiration Month"
              onChange={[Function]}
            />
          </PortaledDraggableListItem>
          <PortaledDraggableListItem
            id="item_31"
            index={5}
            isDraggableViaHandle={true}
            key="5"
          >
            <Checkbox
              isChecked={true}
              label="Country"
              name="Country"
              onChange={[Function]}
            />
          </PortaledDraggableListItem>
          <PortaledDraggableListItem
            id="item_32"
            index={6}
            isDraggableViaHandle={true}
            key="6"
          >
            <Checkbox
              isChecked={true}
              label="State"
              name="State"
              onChange={[Function]}
            />
          </PortaledDraggableListItem>
          <PortaledDraggableListItem
            id="item_33"
            index={7}
            isDraggableViaHandle={true}
            key="7"
          >
            <Checkbox
              isChecked={true}
              label="Function"
              name="Function"
              onChange={[Function]}
            />
          </PortaledDraggableListItem>
        </DraggableList>
      </div>
      <div
        className="column-button-dropdown-footer"
      >
        <PrimaryButton
          onClick={[Function]}
          type="button"
        >
          <FormattedMessage
            defaultMessage="Apply"
            id="boxui.metadataView.applyFiltersButtonText"
          />
        </PrimaryButton>
      </div>
    </div>
  </Overlay>
</Flyout>
`;

exports[`features/metadata-view/components/ColumnButton render should render ColumnButton with template passed in 1`] = `
<Flyout
  className="query-bar-column-dropdown-flyout"
  closeOnClick={true}
  closeOnClickOutside={true}
  closeOnWindowBlur={false}
  constrainToScrollParent={true}
  constrainToWindow={false}
  isVisibleByDefault={false}
  onClose={[Function]}
  onOpen={[Function]}
  openOnHover={false}
  openOnHoverDelayTimeout={300}
  portaledClasses={Array []}
  position="bottom-right"
>
  <Button
    className="query-bar-button"
    isDisabled={false}
    onClick={[Function]}
    type="button"
  >
    <MenuToggle>
      <IconMetadataColumns />
      <span
        className="button-label"
      >
        <FormattedMessage
          defaultMessage="Columns"
          id="boxui.metadataView.columnsButtonText"
        />
      </span>
    </MenuToggle>
  </Button>
  <Overlay>
    <div
      className="column-button-dropdown"
    >
      <div
        className="column-button-dropdown-header"
      >
        <DraggableList
          className="draggable-list-example"
          listId="2"
          onDragEnd={[Function]}
        >
          <PortaledDraggableListItem
            id="item_28"
            index={1}
            isDraggableViaHandle={true}
            key="1"
          >
            <Checkbox
              isChecked={true}
              label="Vendor Name"
              name="Vendor Name"
              onChange={[Function]}
            />
          </PortaledDraggableListItem>
          <PortaledDraggableListItem
            id="item_26"
            index={2}
            isDraggableViaHandle={true}
            key="2"
          >
            <Checkbox
              isChecked={true}
              label="Updated"
              name="Updated"
              onChange={[Function]}
            />
          </PortaledDraggableListItem>
          <PortaledDraggableListItem
            id="item_29"
            index={3}
            isDraggableViaHandle={true}
            key="3"
          >
            <Checkbox
              isChecked={true}
              label="Contract Value"
              name="Contract Value"
              onChange={[Function]}
            />
          </PortaledDraggableListItem>
          <PortaledDraggableListItem
            id="item_30"
            index={4}
            isDraggableViaHandle={true}
            key="4"
          >
            <Checkbox
              isChecked={true}
              label="Expiration Month"
              name="Expiration Month"
              onChange={[Function]}
            />
          </PortaledDraggableListItem>
          <PortaledDraggableListItem
            id="item_31"
            index={5}
            isDraggableViaHandle={true}
            key="5"
          >
            <Checkbox
              isChecked={true}
              label="Country"
              name="Country"
              onChange={[Function]}
            />
          </PortaledDraggableListItem>
          <PortaledDraggableListItem
            id="item_32"
            index={6}
            isDraggableViaHandle={true}
            key="6"
          >
            <Checkbox
              isChecked={true}
              label="State"
              name="State"
              onChange={[Function]}
            />
          </PortaledDraggableListItem>
          <PortaledDraggableListItem
            id="item_33"
            index={7}
            isDraggableViaHandle={true}
            key="7"
          >
            <Checkbox
              isChecked={true}
              label="Function"
              name="Function"
              onChange={[Function]}
            />
          </PortaledDraggableListItem>
        </DraggableList>
      </div>
      <div
        className="column-button-dropdown-footer"
      >
        <PrimaryButton
          onClick={[Function]}
          type="button"
        >
          <FormattedMessage
            defaultMessage="Apply"
            id="boxui.metadataView.applyFiltersButtonText"
          />
        </PrimaryButton>
      </div>
    </div>
  </Overlay>
</Flyout>
`;
